### PR TITLE
DW-258 Added functionality to handle compressed documentOptions

### PR DESCRIPTION
**Pull request by timg-dev**

### CODE REVIEW

File: warehouse/query-core/src/main/java/datawave/query/iterator/FieldIndexOnlyQueryIterator.java [hunk 1]
```diff
@@ -253,7 +253,18 @@ protected void createAndSeekIndexIterator(Range range, Collection<ByteSequence>
         Collection<String> unindexedTypes = Lists.newArrayList();
         
         Set<String> keys = fetchDatatypeKeys(this.documentOptions.get(NON_INDEXED_DATATYPES));
-        unindexedTypes.addAll(keys);
+        
+        String compressedOptionString = this.documentOptions.get(QUERY_MAPPING_COMPRESS);
+        if (compressedOptionString != null) {
```
Comment (apmoriarty) [inline warehouse/query-core/src/main/java/datawave/query/iterator/FieldIndexOnlyQueryIterator.java]: Replace this check with StringUtils.isBlank(). That will catch the case when QUERY_MAPPING_COMPRESS is set to an empty string.

File: warehouse/query-core/src/main/java/datawave/query/iterator/FieldIndexOnlyQueryIterator.java [hunk 2]
```diff
@@ -253,7 +254,18 @@ protected void createAndSeekIndexIterator(Range range, Collection<ByteSequence>
         Collection<String> unindexedTypes = Lists.newArrayList();
         
         Set<String> keys = fetchDatatypeKeys(this.documentOptions.get(NON_INDEXED_DATATYPES));
-        unindexedTypes.addAll(keys);
+        
+        String compressedOptionString = this.documentOptions.get(QUERY_MAPPING_COMPRESS);
```
Comment (ivakegg) [inline warehouse/query-core/src/main/java/datawave/query/iterator/FieldIndexOnlyQueryIterator.java]: Where are we setting these options in a compressed manner.  I would have expected to see compression being done somewhere in concert with this decompression.

**Comment by ivakegg** [comment]
Please resolve conflicts.  sorry.

**Comment by timg-dev** [comment]
Conflicts resolved.

**Pull request merged**